### DISCUSSION TITLE
Can I deploy microservices to Dokploy?

### DISCUSSION THREAD

**Discussion by iamyegor** [Dokploy/dokploy#670] [Q&A] [answered]
Sorry, if the question sounds stupid, I've discovered Dokploy fairly recently. I have Next.js frontend and several ASP.Net Core backends communicating through RabbitMQ, can I deploy all of that using Dokploy?

**Comment by Siumauricio** [comment]
@iamyegor Hi, yes, you can deploy many applications as you want, even I know someone which are running around 60 applications which are mostly microservices, there are no restrictions.

**Reply by iamyegor** [reply]
Can you please give me an advice on how to set up database migrations on this platform? Because currently I'm trying to migrate a database using a dedicated application which is deployed along with my backend and its main purpose is to migrate the database. I'm facing an error with the migrator application restarting every time it finishes the job.

**Reply by Siumauricio** [reply]
Well, there is no restriction on how you should do it, what I would do would be:

Write a docker image, and that image really only executes a command (the migration) once it is initialized and the container dies automatically once the process is finished.

**Reply by iamyegor** [reply]
Thank you for the advice, but the problem is that after the container successfully finishes and dies, dokploy starts a new one.

**Reply by Siumauricio** [reply, marked as answer]
Can you try to assign this config to your application, this will prevent to recreate another container

![Screenshot 2024-11-10 at 4 19 38 PM](https://github.com/user-attachments/assets/83de4cbe-d449-4e62-9c77-5d92a63d7ba6)

**Reply by iamyegor** [reply]
Thank you, that did the job for me. For people who face this problem and wondering where they can find this swarm configuration, you need to:
1. Open your application that is in charge of migrating the database
2. Go to advanced tab, as shown on the picture
![Screenshot 2024-11-11 123255](https://github.com/user-attachments/assets/1c4d53bf-b01f-4ff1-8ce9-4afb5f73e0c3)
3. There, click on the Swarm settings
 ![Screenshot 2024-11-11 123433](https://github.com/user-attachments/assets/3b6a6572-139e-4fb8-9323-0f6b3e1335df)
